flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/mailbox/TaskMailboxImplTest.java

```diff
@@ -225,18 +223,16 @@ private void testUnblocksInternal(
 	/**
 	 * Test producer-consumer pattern through the mailbox in a concurrent setting (n-writer / 1-reader).
 	 */
-	private void testPutTake(
-			FunctionWithException<Mailbox, Runnable, Exception> takeMethod,
-			BiConsumerWithException<Mailbox, Runnable, Exception> putMethod) throws Exception {
+	private void testPutTake(FunctionWithException<Mailbox, Mail, Exception> takeMethod) throws Exception {
 		final int numThreads = 10;
 		final int numLettersPerThread = 1000;
 		final int[] results = new int[numThreads];
 		Thread[] writerThreads = new Thread[numThreads];
 		Mailbox mailbox = taskMailbox.getDownstreamMailbox(DEFAULT_PRIORITY);
 		Thread readerThread = new Thread(ThrowingRunnable.unchecked(() -> {
-			Runnable letter;
-			while ((letter = takeMethod.apply(mailbox)) != POISON_LETTER) {
-				letter.run();
+			Mail mail;
+			while ((mail = takeMethod.apply(mailbox)) != POISON_LETTER) {
+				mail.run();
```

Comment:
I think this is lost in translation. `POISON_LETTER` is of `Runnable`, `mail` is of `Mail`.

Reply:
👍